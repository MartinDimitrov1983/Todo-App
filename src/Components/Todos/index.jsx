import React from 'react';
import PropTypes from 'prop-types';
import { Todo } from '..';

const Todos = ({
  todos,
  resolved,
  unresolved,
  deleteTodoHandler,
  resolveSingleTodoHandler,
  openModal,
  setSelectedTodo,
  selectedColor,
}) => {
  const setActiveTodo = (id) => {
    openModal();
    setSelectedTodo(id);
  };

  const filteredTodos = (todos) => {
    return todos
      .filter((todo) => {
        if (resolved && unresolved) {
          return todo;
        }
        if (resolved) {
          return todo.isResolved === true;
        }
        if (unresolved) {
          return todo.isResolved === false;
        }

        return todo;
      })
      .filter((todo) => {
        if (selectedColor) {
          return todo.color === selectedColor;
        }

        return todo;
      });
  };

  return (
    <>
      {filteredTodos(todos).map((todo, index) => {
        return (
          <Todo
            text={todo.title}
            key={todo.id}
            isResolved={todo.isResolved}
            isDone={todo.isDone}
            onDelete={() => deleteTodoHandler(todo.id)}
            onResolveSingleTodo={(e) => resolveSingleTodoHandler(todo, index)}
            onAddColor={() => setActiveTodo(todo.id)}
            color={todo.color}
          />
        );
      })}
    </>
  );
};

Todos.propTypes = {
  todos: PropTypes.array,
  resolved: PropTypes.bool,
  unresolved: PropTypes.bool,
  deleteTodoHandler: PropTypes.func,
  resolveSingleTodo: PropTypes.func,
  openModal: PropTypes.func,
  setSelectedTodo: PropTypes.func,
  selectedColor: PropTypes.string,
};

export default Todos;
