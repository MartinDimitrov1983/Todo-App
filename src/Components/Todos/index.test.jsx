import React from 'react';
import { render } from '@testing-library/react';

import Todos from './index';

test('Snapshot Todos', () => {
  const { asFragment } = render(
    <Todos
      todos={[
        {
          color: '',
          id: '1-delectus aut autem',
          isDone: false,
          isResolved: true,
          title: 'delectus aut autem',
        },
        {
          color: '',
          id: '2-quis ut nam facilis et officia qui',
          isDone: true,
          isResolved: false,
          title: 'quis ut nam facilis et officia qui',
        },
      ]}
      resolved={false}
      unresolved={false}
      deleteTodoHandler={() => {}}
      resolveSingleTodo={() => {}}
      openModal={() => {}}
      setSelectedTodo={() => {}}
      selectedColor={null}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
