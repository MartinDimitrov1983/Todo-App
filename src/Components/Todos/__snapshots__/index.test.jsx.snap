// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Todos 1`] = `
<DocumentFragment>
  <div
    class="container"
    style="background: rgb(255, 255, 255);"
  >
    <div
      class="text"
    >
      delectus aut autem
    </div>
    <div
      class="checkbox"
    >
      <div
        class="checkBoxWrapper"
      >
        <label
          class="label"
        >
          <input
            checked=""
            class="input"
            type="checkbox"
          />
          <span>
            Resolve
          </span>
        </label>
      </div>
      <div
        class="done"
      >
        Done
      </div>
    </div>
    <div
      class="buttons"
    >
      <button
        class="button"
        type="button"
      >
        <div>
          Add Color
        </div>
      </button>
      <button
        class="button"
        type="button"
      >
        <div>
          Delete
        </div>
      </button>
    </div>
  </div>
  <div
    class="container"
    style="background: rgb(255, 255, 255);"
  >
    <div
      class="text"
    >
      quis ut nam facilis et officia qui
    </div>
    <div
      class="checkbox"
    >
      <div
        class="checkBoxWrapper"
      >
        <label
          class="label"
        >
          <input
            class="input"
            type="checkbox"
          />
          <span>
            Resolve
          </span>
        </label>
      </div>
    </div>
    <div
      class="buttons"
    >
      <button
        class="button"
        type="button"
      >
        <div>
          Add Color
        </div>
      </button>
      <button
        class="button"
        type="button"
      >
        <div>
          Delete
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
